mpkrueger/rollcall-alpha
# Significant refactor of email code.

Easier to test basic email testing.
- additional env.
- TEST_BCC_EMAILS="email1, email2"
- TEST_TO_EMAILS="email1, email2"

Clean Up
- Rewrote the imports to use AWS SDK 3 friendly approach and only require client-ses for efficiency.
- renamed files with LL to not have LL
- renamed the main email function in lib/sendEmail to sendEmail not sendReminderEmail
- removed a lot of unnecessary/old comments

Cron
- updated the cron to point to the right api call

Error Handling
- removed redundant error handling
- lib/sendEmail should create the error and return it back to the 

Notes
signup.js

Asks - pay special attention to...
1. Ensure these code blocks are in the right places.

```
      if (emailSentSuccessfully) {
        await prisma.timeSlot.update({
          where: { id: timeSlot.id },
          data: { reminderSent: true },
        });
      }
```

2. the "Return statements" like below are at the right scope. Need to avoid situations where a function ends up not returning something.
`return res.status(200).json({ message: res.message });`

TODO.

- remove the hardcoded emails from the sendTestEmail 
- centralize email template html
- refactor and modularize managing separate types of email
- add tests
- make it easier to test each type of email
- more clean up needed and removal of stale comments
- add nodemailer rate limiting logic for fun
